partner_lastname_uppercase/models/res_partner.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Coop IT Easy SC
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model
+    def uppercase_all_lastnames(self):
+        individuals = self.search(
+            [
+                ("is_company", "=", False),
+                ("lastname", "!=", False),
+            ]
+        )
+        for partner in individuals:
+            partner.lastname = partner.lastname.upper()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            is_company = vals.get("is_company", False)
+            lastname = vals.get("lastname", False)
+            if lastname and not is_company:
+                vals["lastname"] = lastname.upper()
+        return super().create(vals_list)
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "lastname" in vals:
+            individuals = self.filtered(lambda rp: rp.lastname and not rp.is_company)
+            for partner in individuals:
+                super(ResPartner, partner).write({"lastname": partner.lastname.upper()})
```

Comment:
at first glance, this way of doing seemed a little confusing to me (writing first then upper-casing). why not simply upper-casing the value in `vals`? i understand that it is because companies should be excluded, but a comment would be useful.

also, is it useful to test for `rp.lastname` while we know that we just wrote to it? oh ok, it is to allow to set it to `False`? a comment would be useful also. ☺️